nexus/db-model/src/inventory.rs

```diff
@@ -1164,32 +1399,29 @@ impl From<InvDataset> for nexus_types::inventory::Dataset {
     }
 }
 
-/// Information about a sled's Omicron zones, part of
-/// [`nexus_types::inventory::SledAgent`].
-///
-/// TODO: This table is vestigial and can be combined with `InvSledAgent`. See
-/// [issue #6770](https://github.com/oxidecomputer/omicron/issues/6770).
+/// Top-level information contained in an [`OmicronSledConfig`].
 #[derive(Queryable, Clone, Debug, Selectable, Insertable)]
-#[diesel(table_name = inv_sled_omicron_zones)]
-pub struct InvSledOmicronZones {
+#[diesel(table_name = inv_omicron_sled_config)]
+pub struct InvOmicronSledConfig {
     pub inv_collection_id: DbTypedUuid<CollectionKind>,
-    pub time_collected: DateTime<Utc>,
-    pub source: String,
-    pub sled_id: DbTypedUuid<SledKind>,
+    pub id: Uuid,
     pub generation: Generation,
+    pub remove_mupdate_override: Option<DbTypedUuid<MupdateOverrideKind>>,
 }
 
-impl InvSledOmicronZones {
+impl InvOmicronSledConfig {
     pub fn new(
         inv_collection_id: CollectionUuid,
-        sled_agent: &nexus_types::inventory::SledAgent,
-    ) -> InvSledOmicronZones {
-        InvSledOmicronZones {
+        id: Uuid,
+        config: &OmicronSledConfig,
```

Comment:
Wonder if this should pass in the generation and remove mupdate override UUIDs separately. Currently a full `OmicronSledConfig` structure would have to be constructed which is a lot more heavy-duty. (To me it would also suggest that `InvOmicronSledConfig` stores all sled config, not just the generation and remove_mupdate_override fields.)